__tests__/app.test.js

```diff
@@ -72,79 +69,165 @@ describe("app", () => {
         return request(app)
-        .get("/api/articles/100/comments")
-        .expect(404)
-        .then(({body}) => {
-          expect(body.msg).toBe("Not Found")
-        })    
-      })
+          .get("/api/articles/100/comments")
+          .expect(404)
+          .then(({ body }) => {
+            expect(body.msg).toBe("Not Found");
+          });
+      });
       test("GET 400: should return with an error if an invalid file path is given", () => {
         return request(app)
-        .get("/api/articles/invalid/comments")
-        .expect(400)
-        .then(({body}) => {
-          expect(body.msg).toBe("Bad Request")
-        })
+          .get("/api/articles/invalid/comments")
+          .expect(400)
+          .then(({ body }) => {
+            expect(body.msg).toBe("Bad Request");
+          });
+      });
+    });
+  });
+  describe("POST test", () => {
+    describe("/api/articles/:article_id/comments", () => {
+      test("POST 200: should accept a valid comment post request to a specific article (i.e. article_id). It should respond with the posted comment", () => {
+        return request(app)
+          .post("/api/articles/1/comments")
+          .send({
+            body: "excellent work",
+            username: "butter_bridge",
+          })
+          .expect(201)
+          .then(({ body }) => {
+            expect(body.addedComment).toEqual({
+              article_id: 1,
+              body: "excellent work",
+              votes: 0,
+              author: "butter_bridge",
+              created_at: String(new Date(Date.now())),
+              comment_id: 19,
+            });
+          });
+      });
+      test("POST 400: should respond with an error if there are missing keys in the post", () => {
+        return request(app)
+          .post("/api/articles/1/comments")
+          .send({
+            username: "butter_bridge",
+          })
+          .expect(400)
+          .then(({ body }) => {
+            expect(body.msg).toBe("Bad Request");
+          });
+      });
+      test("POST 400: should return with an error if an invalid username is given", () => {
+        return request(app)
+          .post("/api/articles/1/comments")
+          .send({
+            username: "alex",
+            body: "excellent work",
+          })
+          .expect(400)
+          .then(({ body }) => {
+            expect(body.msg).toBe("Bad Request");
+          });
+      });
+      test("POST 400: should return an error if body datatype is number", () => {
+        return request(app)
+          .post("/api/articles/1/comments")
+          .send({
+            username: "butter_bridge",
+            body: 500,
+          })
+          .expect(400)
+          .then(({ body }) => {
+            expect(body.msg).toBe("Bad Request");
+          });
+      });
+      test("POST 400: should return an error if an invalid file path is given", () => {
```

Comment:
We can be a bit more specific with the description here, it's actually that the article_id is invalid in this case.